source/parameter_acceptor.cc

```diff
@@ -113,26 +115,68 @@ ParameterAcceptor::log_info()
   deallog.pop();
 }
 
-
 void ParameterAcceptor::parse_all_parameters(ParameterHandler &prm)
 {
+  std::string sec="";
+  std::string subsec="";
   for (unsigned int i=0; i< class_list.size(); ++i)
     if (class_list[i] != NULL)
       {
-        prm.enter_subsection(class_list[i]->get_section_name());
```

Comment:
All of this could be done without limiting to one subsection by 
~~~
secs = Utilities::split_string_list(class_list[i]->get_section_name(), sep);
for(unsigned int i=0; i<secs.size(); ++i) {
   Assert(secs[i] != "", ExcMessage("Cannot enter empty subsection!"));
  prm.enter_subsection(secs[i]);
}

....
for(unsigned int i=0; i<secs.size(); ++i)
  prm.leave_subsection();
~~~